### PR TITLE
Fix/Improve Selects

### PR DESCRIPTION
**Pull request by Hypercubed**
## Summary

* Fix issues where fill style not applied in released code
* Show active state when focused
* Keep focus state after selection
* Enable more keyboard controls (see below)
* Skip focus on disabled inputs
* Skip focus on disable and hidden options
* Add roles attribute
* Add for attribute to label

## Checklist

- [ ] \*Added unit tests
- [x] \*Added a code reviewer
- [x] Added changes to `/projects/swimlane/ngx-ui/CHANGELOG.md` under HEAD (Unreleased)
- [x] Updated the demo page
- [ ] Included screenshots of visual changes

_\*required_

keyboard controls:

* Tab to focus on select element

* When Focused and closed:
  * `Space` or `Downarrow` to open dropdown and highlight first item
  * `Uparrow` to open dropdown and highlight last item

* When dropdown is open:
  * `Downarrow` to highlight next item (skipping disabled or hidden)
  * `Uparrow` to highlight prev item (skipping disabled or hidden)
  * `Space` or `Enter` to select highlighted item
  *  `Esc` to close dropdown

* When tagging Tab focuses on input
  * `Backspace` to remove last tag.

### CODE REVIEW

File: projects/swimlane/ngx-ui/src/lib/components/select/select-dropdown.component.ts [hunk 1]
```diff
@@ -200,18 +175,52 @@ export class SelectDropdownComponent implements AfterViewInit {
...
-    const key = event.key;
+    switch (event.code) {
+      case KeyboardKeys.ESCAPE:
+        return this.close.emit(true);
+      case KeyboardKeys.ARROW_DOWN:
+        return this.focusNext();
+      case KeyboardKeys.ARROW_UP:
+        return this.focusPrev();
+      case KeyboardKeys.ENTER:
+      case KeyboardKeys.SPACE:
```
Comment (marjan-georgiev) [inline projects/swimlane/ngx-ui/src/lib/components/select/select-dropdown.component.ts:194]: Does this affect the filter input?

File: projects/swimlane/ngx-ui/src/lib/components/select/select-input.component.ts [hunk 2]
```diff
@@ -138,45 +111,64 @@ export class SelectInputComponent implements AfterViewInit, OnChanges {
...
 
+  // Events on ngx-select-input-box element
   onGlobalKeyUp(event: KeyboardEvent) {
     event.stopPropagation();
-    const key = event.key;
 
-    if (key === KeyboardKeys.ARROW_DOWN) {
-      this.activate.emit(event);
+    switch (event.code) {
+      case KeyboardKeys.SPACE:
```
Comment (marjan-georgiev) [inline projects/swimlane/ngx-ui/src/lib/components/select/select-input.component.ts:149]: Same as above, will this execute if space is pressed in the filter input?

File: projects/swimlane/ngx-ui/src/lib/components/select/select-dropdown.component.ts (hunk 1 above)
Comment (Hypercubed) [reply projects/swimlane/ngx-ui/src/lib/components/select/select-dropdown.component.ts:203]: Should not... the input has a different event listeners (now named `onInputKeyUp` and `onInputKeyDown`)

Comment (Hypercubed) [reply projects/swimlane/ngx-ui/src/lib/components/select/select-dropdown.component.ts:203]: Oh.. that's the other component... but same answer.  Doesn't impact AFAICT.

File: projects/swimlane/ngx-ui/src/lib/components/select/select-dropdown.component.html [hunk 3]
```diff
@@ -44,8 +44,8 @@
           [class.active]="kv.index === focusIndex"
           [class.selected]="isSelected(kv.option)"
           [hidden]="kv.option.hidden"
-          tabindex="-1"
-          (click)="selection.emit(kv.option)"
+          [attr.tabindex]="-1"
```
Comment (Hypercubed) [inline projects/swimlane/ngx-ui/src/lib/components/select/select-dropdown.component.html:47]: This is not needed... will remove.

File: projects/swimlane/ngx-ui/src/lib/components/select/select.component.scss [hunk 4]
```diff
@@ -571,13 +572,9 @@ $color-chip-text: #fff;
...
 
-    &.single-selection .ngx-select-input .ngx-select-clear {
-      top: 50%;
-      z-index: 500;
-    }
-
+    &.single-selection .ngx-select-input .ngx-select-clear,
     .ngx-select-caret {
-      top: 50%;
+      top: 1.5em;
```
Comment (Hypercubed) [inline projects/swimlane/ngx-ui/src/lib/components/select/select.component.scss:577]: Doesn't look right for tagging... will address.

File: cypress/integration/forms/selects.spec.ts [hunk 5]
```diff
@@ -45,11 +68,183 @@ describe('Selects', () => {
...
+        cy.realPress('Tab');
+        shouldBeNotFocused();
+        shouldBeNotActive();
+      });
+    });
+  });
+
+  describe('Tagging', () => {
+    beforeEach(() => {
+      cy.get('#section-3').first().as('SUT');
```
Comment (Hypercubed) [inline cypress/integration/forms/selects.spec.ts:159]: This should be by label or data-cy... will update.

**Review by marjan-georgiev — APPROVED** [review]

**Pull request merged**